### PR TITLE
Allow compiler language selection via query parameter, fixes #1489

### LINKED ISSUE #1489: Allow switching between Solidity and Yul via a URL parameter
Related to https://github.com/ethereum/solidity/issues/11415#issuecomment-891105568

Remix allows creating links with embedded code snippets using `code` query parameter, e.g. https://remix.ethereum.org/?code=Y29udHJhY3QK. Even though Remix supports both Solidity and Yul, such links always open the editor with Soldity selected. It would be useful to be able to specify the language in the URL.

### PR DESCRIPTION
**Pull request by rorye**
Fixes #1489 

Example usage:

http://localhost:8080/#optimize=false&runs=200&evmVersion=null&language=Yul&version=soljson-v0.8.4+commit.c7e474f2.js&code=b2JqZWN0ICJNeUNvbnRyYWN0IiB7CiAgY29kZSB7CiAgICAvLyB0aGlzIGlzIHRoZSBjb25zdHJ1Y3Rvci4KICAgIC8vIHN0b3JlIHRoZSBjcmVhdG9yIGluIHRoZSBmaXJzdCBzdG9yYWdlIHNsb3QKICAgIHNzdG9yZSgwLCBjYWxsZXIoKSkKICAgIC8vIG5vdyByZXR1cm4gdGhlIHJ1bnRpbWUgY29kZSB1c2luZyB0aGUgc3BlY2lhbCBmdW5jdGlvbnMKICAgIGRhdGFjb3B5KDAsIGRhdGFvZmZzZXQoIlJ1bnRpbWUiKSwgZGF0YXNpemUoIlJ1bnRpbWUiKSkKICAgIHJldHVybigwLCBkYXRhc2l6ZSgiUnVudGltZSIpKQogIH0KICBvYmplY3QgIlJ1bnRpbWUiIHsKICAgIGNvZGUgewogICAgICAvLyBydW50aW1lIC0ganVzdCByZXR1cm4gdGhlIGNyZWF0b3IKICAgICAgbXN0b3JlKDAsIHNsb2FkKDApKQogICAgICByZXR1cm4oMCwgMHgyMCkKICAgIH0KICB9Cn0K

Screen Shot 2021-08-22 at 9 52 07 am

### CODE REVIEW

File: libs/remix-ui/solidity-compiler/src/lib/compiler-container.tsx [hunk 1]
```diff
@@ -77,7 +78,8 @@ export const CompilerContainer = (props: CompilerContainerProps) => {
           includeNightlies: api.getConfiguration('includeNightlies') || false,
           optimise: (optimize !== null) && (optimize !== undefined) ? optimize : api.getConfiguration('optimise') || false,
           runs: (runs !== null) && (runs !== 'null') && (runs !== undefined) && (runs !== 'undefined') ? runs : 200,
-          evmVersion: (evmVersion !== null) && (evmVersion !== 'null') && (evmVersion !== undefined) && (evmVersion !== 'undefined') ? evmVersion : 'default'
+          evmVersion: (evmVersion !== null) && (evmVersion !== 'null') && (evmVersion !== undefined) && (evmVersion !== 'undefined') ? evmVersion : 'default',
+          language: (language !== null) && (language !== 'null') && (language !== undefined) && (language !== 'undefined') ? language : 'Solidity'
```
Comment (LianaHus) [inline libs/remix-ui/solidity-compiler/src/lib/compiler-container.tsx:82]: what if it is not 'Yul'?

File: libs/remix-ui/solidity-compiler/src/lib/compiler-container.tsx [hunk 2]
```diff
@@ -526,8 +528,8 @@ export const CompilerContainer = (props: CompilerContainerProps) => {
           <div className="mb-2">
             <label className="remixui_compilerLabel form-check-label" htmlFor="compilierLanguageSelector">Language</label>
             <select onChange={(e) => handleLanguageChange(e.target.value)} value={state.language} className="custom-select" id="compilierLanguageSelector" title="Available since v0.5.7">
-              <option value='Solidity'>Solidity</option>
-              <option value='Yul'>Yul</option>
+              <option data-id={state.language === 'Solidity' ? 'selected' : ''} value='Solidity'>Solidity</option>
```
Comment (LianaHus) [inline libs/remix-ui/solidity-compiler/src/lib/compiler-container.tsx:531]: we could handle lower cases as well.

**Review by LianaHus — CHANGES_REQUESTED** [review]

File: libs/remix-ui/solidity-compiler/src/lib/compiler-container.tsx (hunk 1 above)
Comment (rorye) [reply libs/remix-ui/solidity-compiler/src/lib/compiler-container.tsx:82]: So we should ensure that it's either `Yul` or `Solidity` ?

File: libs/remix-ui/solidity-compiler/src/lib/compiler-container.tsx (hunk 2 above)
Comment (rorye) [reply libs/remix-ui/solidity-compiler/src/lib/compiler-container.tsx:531]: Thanks, will take a look.

**Comment by ioedeveloper** [comment]
@rorye Can you update and resolve merge conflicts on your PR so we can proceed?

**Comment by yann300** [comment]
@ioedeveloper this one is ready for review and merge, i've added some commits to it

**Review by ioedeveloper — APPROVED** [review]

**Pull request merged**